### PR TITLE
[4.0] Use Custom Element tabs on Global Config

### LINKED ISSUE #25876: [4.0] [backend-template] Tabs styling
### Steps to reproduce the issue

1. Go to create an new article
2. Look at the styling of the blue tabs

tabs1

1. Go to a component options page
2. Look at the styling of the tabs

tabs2

### Expected result

Same styling

### Actual result

Different styling

### Additional comments

One uses Bootstrap tabs and the other uses Custom Elements

### PR DESCRIPTION
**Pull request by C-Lodder**
### Fixes
#25876 

### Summary of Changes

Replace the Bootstrap tabs on the Global Config page with Custom Elements.

### Testing Instructions

1. Go to `administrator/index.php?option=com_config`
2. Navigate through the tabs and ensure they display the correct content

### CODE REVIEW

**Comment by brianteeman** [comment]
tested and it works but you also need to do the tabs in ?option=com_config&view=component

**Comment by C-Lodder** [comment]
Yeah just noticed. Trying to sort it now.

**Comment by C-Lodder** [comment]
@brianteeman think it's done now. Not sure if anything is missing cause the view was a mess, beyond belief.

**Comment by brianteeman** [comment]
1. think you got it everywhere now
2. check the permissions tab as it now has a big indent again
image

3. on a page with a lot of tabs is it supposed to have a scroll bar?
image

**Comment by C-Lodder** [comment]
2. The permissions tabs are another issues. I'll fix them in this PR so it's not forgotten about.
3. Not sure about the scrollbar, but nothing to do with this PR.

**Comment by brianteeman** [comment]
Well the scrollbar didn't appear with the b's tabs

**Comment by C-Lodder** [comment]
All I'm doing it using the Custom Elements for the tabs. If there are styling issues, they need to be fixed in the CE repo.

A lot of tabs would result in a scrollbar in other pages that are already using the tabs, not just this page

**Comment by C-Lodder** [comment]
Spacing is there due to it being rendered with `renderFieldset()` which wraps it inside `<div class="controls">`. Not sure if there is another method to render it.

**Comment by richard67** [comment]
>     2. The permissions tabs are another issues. I'll fix them in this PR so it's not forgotten about.

@C-Lodder So we shall wait with testing this PR until you've added that, too?

**Comment by dgrammatiko** [comment]
@C-Lodder please add the todo for the `showon` as https://github.com/joomla/joomla-cms/pull/23780#pullrequestreview-205301544

**Comment by vaibhavsTekdi** [comment]
I have tested this item :white_check_mark: successfully on 1ed6b477c53c93a5a9706325ceaaddc14e62d67d<br /><br />I have tested this successfully. #jpbf19 #test1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26890">issues.joomla.org/tracker/joomla-cms/26890</a>.</sub>

**Comment by ttplpoojak** [comment]
I have tested this item :white_check_mark: successfully on 1ed6b477c53c93a5a9706325ceaaddc14e62d67d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26890">issues.joomla.org/tracker/joomla-cms/26890</a>.</sub>

**Comment by dgrammatiko** [comment]
> Well the scrollbar didn't appear with the b's tabs

The scrollbars are part of the tabs design (but should be visible only on smaller screens). The idea comes from https://bulma.io/documentation/components/tabs/

**Comment by C-Lodder** [comment]
@mbabker do you know of another way to render the permissions element without renderFieldSet() ?

File: administrator/components/com_config/tmpl/component/default.php [hunk 1]
```diff
@@ -55,97 +55,59 @@
...
 
-					<?php if (!$isGrandchild && $hasParent) : ?>
-						<fieldset id="fieldset-<?php echo $this->escape($name); ?>" class="options-grid-form options-grid-form-full">
-							<legend><?php echo Text::_($fieldSet->label); ?></legend>
-							<div>
-					<?php elseif (!$hasParent) : ?>
-						<?php if ($opentab) : ?>
 
-							<?php if ($opentab > 1) : ?>
+					<fieldset id="fieldset-<?php echo $this->escape($name); ?>" class="options-grid-form options-grid-form-full">
```
Comment (Quy) [inline administrator/components/com_config/tmpl/component/default.php]:
```suggestion
					<fieldset id="fieldset-<?php echo $this->escape($name); ?>" class="options-grid-form options-grid-form-full options-menu">
```

Comment (Quy) [reply administrator/components/com_config/tmpl/component/default.php]: This fixes the indentation.

**Comment by C-Lodder** [comment]
@Quy Thanks for the commit. Much appreciated

@brianteeman would you mind giving it another spin please?

**Comment by brianteeman** [comment]
@C-Lodder I am away this weekend without a laptop.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on a29d7fdb42ed09f593544e176603af9d1d42c807<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26890">issues.joomla.org/tracker/joomla-cms/26890</a>.</sub>

**Comment by C-Lodder** [comment]
@brianteeman just a reminder if you find any time to do a quick test

**Comment by brianteeman** [comment]
Sorry. Completely forgot

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 9d084102ce4ad9e8087c4c9d529a1e732744ba3f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26890">issues.joomla.org/tracker/joomla-cms/26890</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 9d084102ce4ad9e8087c4c9d529a1e732744ba3f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26890">issues.joomla.org/tracker/joomla-cms/26890</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26890">issues.joomla.org/tracker/joomla-cms/26890</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Comment by ghazal** [comment]
I have tested this item :red_circle: unsuccessfully on 29f07de76a4bfad8c38a917d5133ade2e1ea6662<br /><br />Testing this on last Nightly Build (ie 25/11).<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26890">issues.joomla.org/tracker/joomla-cms/26890</a>.</sub>

global-config

**Comment by wilsonge** [comment]
@ghazal given this got merged last night can you create a fresh issue for your bug so that it can be tracked properly - thanks!

**Comment by C-Lodder** [comment]
@wilsonge @ghazal The overlapping has nothing to do with this PR. The fields inside the parent should respond to the width of the parent.

I have already raised an issue about the poor UX with fieldsets.....the fact the labels are to the left of the field and the sets are spanned horizontally.

**Comment by ghazal** [comment]
@C-Lodder 
Should I let you create the PR ? 
Most probably your description will be more detailed than mine.
@wilsonge

**Comment by C-Lodder** [comment]
@ghazal I think the powers that be seem to be happy with the current horizontal layout. I'll let the issue I raised speak for itself once there's more feedback on it: https://github.com/joomla/joomla-cms/issues/25891

**Comment by richard67** [comment]
@C-Lodder Fact is that the global config fields are more broken now as before, like @ghazal 's screenshot shows in his [comment above](https://github.com/joomla/joomla-cms/pull/26890#issuecomment-558124766). This is not really obvious in #25891 , so I think either it needs a new screenshot in the description of #25891so people can see it deals also with the issue shown by @ghazal , or someone has to create a new issue, so people can see it is known. If new issue and nobody has the time to create one: I can do that, just let me know.

**Pull request merged**